laplace/lllaplace.py

```diff
@@ -134,18 +142,28 @@ def _glm_predictive_distribution(self, X, joint=False):
 
         return (f_mu.detach(), f_var.detach()) if not self.enable_backprop else (f_mu, f_var)
 
-    def _nn_predictive_samples(self, X, n_samples=100):
+    def _nn_predictive_samples(self, X, n_samples=100, generator=None, **model_kwargs):
         fs = list()
-        for sample in self.sample(n_samples):
+        for sample in self.sample(n_samples, generator):
             vector_to_parameters(sample, self.model.last_layer.parameters())
-            f = self.model(X.to(self._device))
+            f = self.model(X.to(self._device), **model_kwargs)
             fs.append(f.detach() if not self.enable_backprop else f)
         vector_to_parameters(self.mean, self.model.last_layer.parameters())
         fs = torch.stack(fs)
         if self.likelihood == 'classification':
             fs = torch.softmax(fs, dim=-1)
         return fs
 
+    def _nn_predictive_classification(self, X, n_samples=100, generator=None, **model_kwargs):
+        py = 0
+        for sample in self.sample(n_samples, generator):
+            vector_to_parameters(sample, self.model.last_layer.parameters())
+            # TODO: Implement with a single forward pass until last layer.
```

Comment:
Maybe we should do this now, since it will make predictions a lot faster for large foundation models. Should be possible to implement with `forward_with_features()`?

Reply:
Seems to be nontrivial at first glance. Let's do this in a separate PR.